src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -125,6 +130,21 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
         }
 
         codeDocument.SetDocumentIntermediateNode(document);
+
+        static bool TryRemoveGlobalPrefixFromDefaultUsing(in UsingReference usingReference, out UsingReference trimmed)
```

Comment:
Did you consider `out Span<char> trimmed` to avoid the allocation then changing the `Contains` call above to one that can search against `Span<char>`?